core/src/main/java/nl/nn/adapterframework/doc/model/ElementRole.java

```diff
@@ -16,22 +16,47 @@
 
 package nl.nn.adapterframework.doc.model;
 
+import java.util.Collection;
+import java.util.Comparator;
 import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
+import org.apache.logging.log4j.Logger;
+
+import lombok.AccessLevel;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import lombok.Setter;
 import nl.nn.adapterframework.doc.Utils;
+import nl.nn.adapterframework.util.LogUtil;
+
+public class ElementRole implements Comparable<ElementRole> {
+	private static Logger log = LogUtil.getLogger(ElementRole.class);
 
-public class ElementRole {
-	private final @Getter ElementType elementType;
+	private @Getter ElementType elementType;
 	private final @Getter String syntax1Name;
 	private final int syntax1NameSeq;
+	private @Getter boolean deprecated;
+	private @Getter @Setter(AccessLevel.PACKAGE) boolean superseded;
 
-	private ElementRole(ElementType elementType, String syntax1Name, int syntax1NameSeq) {
+	private ElementRole(ElementType elementType, String syntax1Name, int syntax1NameSeq, boolean isDeprecated) {
 		this.elementType = elementType;
 		this.syntax1Name = syntax1Name;
 		this.syntax1NameSeq = syntax1NameSeq;
+		this.deprecated = isDeprecated;
+		this.superseded = false;
+	}
+
+	void updateDeprecated(boolean newDeprecated) {
+		if(deprecated != newDeprecated) {
+			log.warn(String.format("Ambiguous deprecated status of ElementRole [%s], set to false", toString()));
+		}
```

Comment:
Deze amiguiteit is het gevolg van dat deprecated status niet van de ElementRole is, maar van de koppeling van de ElementRole aan een FrankElement.

Reply:
Dat klopt, maar ik moet een keuze maken of ik een ElementRole als deprecated beschouw. Dat veld gebruik ik vervolgens om  conflicten op te lossen.